sys/parsoid.js

```diff
@@ -528,14 +530,46 @@ function replaceSections(original, sectionsJson) {
             }
         });
     }
+
+    function sliceSection(offset) {
+        return originalBody.substring(offset.html[0], offset.html[1]);
+    }
+
     sectionIds.sort(function(id1, id2) {
         return sectionOffsets[id2].html[0] - sectionOffsets[id1].html[0];
     })
     .forEach(function(id) {
-        var offset = sectionOffsets[id];
-        newBody = newBody.substring(0, offset.html[0])
-        + sectionsJson[id]
-        + newBody.substring(offset.html[1], newBody.length);
+        var sectionReplacement = sectionsJson[id].html;
+        if (typeof sectionReplacement === 'string') {
```

Comment:
I think we can drop strings completely, and always require an array (or undefined) for `html`.